### DISCUSSION TITLE
Suggestions for the widget's UI design

### DISCUSSION THREAD

**Discussion by pdimu** [aleksey-saenko/MusicRecognizer#63] [Ideas]
Compared to Shazam, Audile stays the same when the widget is resized

![Screenshot_2024-06-22-19-55-37-781_app.lawnchair.jpg](https://github.com/aleksey-saenko/MusicRecognizer/assets/128509556/92249ce5-e4b4-45bf-bd95-5cd3a652cca5)
![Screenshot_2024-06-22-19-56-21-824_app.lawnchair.jpg](https://github.com/aleksey-saenko/MusicRecognizer/assets/128509556/260ccacd-efbe-401e-8e5d-0955a02f6b74)
![Screenshot_2024-06-22-19-56-55-494_app.lawnchair.jpg](https://github.com/aleksey-saenko/MusicRecognizer/assets/128509556/1a3e46e1-00c3-41c5-9a7d-cfa4a9f0180b)
![Screenshot_2024-06-22-19-57-03-427_app.lawnchair.jpg](https://github.com/aleksey-saenko/MusicRecognizer/assets/128509556/09659a29-13bb-439c-80da-be08250133a9)

**Comment by aleksey-saenko** [comment]
Actually, the widget uses different layouts depending on the size, but this is more noticeable when displaying a track. Although I agree that a more beautiful layout could be devised, in my opinion, Shazam doesn't look much better.
<img src='https://github.com/aleksey-saenko/MusicRecognizer/assets/109096599/c71e3a72-e330-44a4-91f0-a58f9bc02bd3' width='150'> <img src='https://github.com/aleksey-saenko/MusicRecognizer/assets/109096599/5bb78512-f0a3-43f3-851a-6bb28a749104' width='150'> <img src='https://github.com/aleksey-saenko/MusicRecognizer/assets/109096599/e1b5916e-01c7-4a20-b562-8d43636587b5' width='150'> <img src='https://github.com/aleksey-saenko/MusicRecognizer/assets/109096599/bd5f5aba-530a-4200-8d7a-82ec89ba03e5' width='150'>

**Comment by pdimu** [comment]
Do you have an idea on how to redesign the widget?
It can use the Shazam one, but when there is a track found, it goes smaller (the waveform icon)

**Comment by aleksey-saenko** [comment]
Shazam always displays the track information in a row. In this case, it's not possible to use big artwork image, since we need enough space for the text. This creates a significant amount of unused vertical space, which Shazam fills with a large button for recognition. Making this button smaller wouldn't solve the space issue.

Displaying the track information in a column (as in Audile) but with a small recognition button at the top seems not good to me.

In my view, the current version is better than Shazam. I'll move this topic to discussions since it's not a real issue; it's more about personal taste. It would be nice to get more opinions or suggestions with images on this matter.

**Comment by zachvlat** [comment]
I 'd like to suggest an alternative size for this widget to be more minimalistic (I 'm not a designer sorry for my bad job!)

<img src='https://github.com/user-attachments/assets/f64956cf-2b19-4e6b-a663-5af40e39a760' width='200'>

**Reply by aleksey-saenko** [reply]
It's a valid suggestion, I think this could be implemented in future versions.

**Reply by aleksey-saenko** [reply]
Something like this?
[wgt.webm](https://github.com/user-attachments/assets/1b2f7c44-8fe6-4f1b-b0e7-8bea1773dd3d)

**Reply by pdimu** [reply]
that looks perfect

**Reply by zachvlat** [reply]
looks very good!
